### PR TITLE
fix Sprite/Label/FastTMX auto-culling bug.

### LINKED ISSUE #10487: Camera fails to show sprites outside visible size
From the forum:
http://discuss.cocos2d-x.org/t/camera-bug-when-moving/19904/2

Using 3.4 final, I create a bunch of sprites, then position them one by one to the right.  I move the camera to the right, and everything off the screen is hidden.  However, if I move the layer to the left, I can see the sprites.

Code from forum post:

The init of HelloWorld:

```
auto spritebatch = SpriteBatchNode::create("MySprites.png");
addChild(spritebatch);
auto spriteCache = SpriteFrameCache::getInstance();
spriteCache->addSpriteFramesWithFile("MySprites.plist");
```

Then I create the sprites and add them to the layer, each one positioned to the right of the previous sprite.

```
for (int i = 0; i < 100; i++) {
    auto sprite = Sprite::createWithSpriteFrameName("MySprite.png");
    sprite->setPosition(Vec2(visibleSize.width/2 + origin.x + (visibleSize.width * i), visibleSize.height/2 + origin.y));
    this->addChild(sprite);
}
```

In the onEnter:

```
void HelloWorld::onEnter()
{
Layer::onEnter();
Size visibleSize = Director::getInstance()->getVisibleSize();
Camera *camera = Director::getInstance()->getRunningScene()->getCameras().front();
auto move = MoveBy::create(35.0, Vec3(10 * visibleSize.width, 0, 0));
camera->runAction(move);
}
```

Related?
https://github.com/cocos2d/cocos2d-x/issues/10275

### PR DESCRIPTION
**Pull request by fusijie**
To fix #10487.

### CODE REVIEW

**Comment by pandamicro** [comment]
@dabingnn Please review

File: cocos/2d/CCCamera.cpp [hunk 1]
```diff
@@ -539,4 +539,10 @@ int Camera::getRenderOrder() const
     return result;
 }
 
+void Camera::visit(Renderer* renderer, const Mat4 &parentTransform, uint32_t parentFlags)
```
Comment (dabingnn) [inline cocos/2d/CCCamera.cpp:542]: There may be some bug if the camera is visited after other objects (such as sprite etc). In this case the update of `_viewProjectionUpdated` is too late.

Comment (fusijie) [inline cocos/2d/CCCamera.cpp:542]: It will lead to display one frame display.

**Comment by dabingnn** [comment]
@pandamicro This pull request could be merged, it provides a temp solution for multiple camera rendering culling while do not affect the performance of 2D situations.

**Pull request merged**